change config to allow exporting of multiple generators and refiners

LoganBarnett opened the pull request:
The configuration has changed such that a module can export multiple generators, which in turn emit multiple refiners all within the same file.

Document an example of using `degenString`, and add documentation for creating a custom generator.

This configuration change is breaking!

LoganBarnett committed:
change config to allow exporting of multiple generators and refiners

The configuration has changed such that a module can export multiple generators,
which in turn emit multiple refiners all within the same file.

Document an example of using degenString, and add documentation for creating a
custom generator.

This configuration change is breaking!

LoganBarnett commented:
@gyrfalcon

gyrfalcon commented on src/config-generator.js:
```diff
@@ -10,22 +10,33 @@ import {
+  degenField('generators', degenList(degenObject(configGeneratorType, [
+    degenField('exports', degenMapping(degenString(), degenString())),
+    degenField('inputFile', degenString()),
+    degenField('outputFile', degenString()),
```
This may just be a minor item, since `degenFilePath` is the same as `degenString`, but if in the future `degenFilePath` is smart about refining paths, there will have been a loss of specificity in the spec.

LoganBarnett replied:
Changed to use `degenFilePath` where appropriate!

gyrfalcon commented:
Aside from the one (potentially minor) thing I called out, looks good!

LoganBarnett committed:
use file paths for parts of the config that refer to paths

This should have no material change but could be used with opaque types at some
point. It also communicates intent better, which is perhaps the best reason to
use degenFilePath.
